ironfist/src/cpp/combat/combat_spells.cpp

```diff
@@ -137,4 +138,25 @@ void combatManager::Resurrect(int spell, int hex, int spellpower) {
 	}
 	this->DrawFrame(1, 0, 0, 0, 75, 1, 1);
 	creat->creature.creature_flags &= ~DEAD;
-}
\ No newline at end of file
+}
+
+float army::SpellCastWorkChance(int spell)
+{
+	if (this->effectStrengths[EFFECT_ANTI_MAGIC]
+		|| this->creature.creature_flags == DEAD
+		&& spell != SPELL_RESURRECT
+		&& spell != SPELL_RESURRECT_TRUE
+		&& spell != SPELL_ANIMATE_DEAD
+		|| this->dead
+		|| this->creatureIdx == CREATURE_GREEN_DRAGON
+		|| this->creatureIdx == CREATURE_RED_DRAGON
+		|| this->creatureIdx == CREATURE_BLACK_DRAGON)
+		return 0.0;
+	if (this->creatureIdx == CREATURE_EARTH_ELEMENTAL

```

Comment:
Nitpick: I'd recommend leaving a blank line between if statements, to make it easier for the eye to tease them apart.